openlane/scripts/klayout/render.py

```diff
@@ -44,108 +44,78 @@
-            ):
-                value = os.path.abspath(value)
-
-            args.append(f"{key}={value or 'NULL'}")
-
-        os.execlp("klayout", *args)
-
-    if __name__ == "__main__":
-        cli()
-
-
-if TYPE_CHECKING:
-    # Dummy data for type-checking
-    input: str = ""
-    output: str = ""
-    input_lefs: str = ""
-    lyp: str = ""
-    lyt: str = ""
-    lym: str = ""
-
-try:
-    gds = input.endswith(".gds")
-
-    # Load technology file
-    tech = pya.Technology()
-    tech.load(lyt)
-
-    layout_options = None
-    if not gds:
-        layout_options = tech.load_layout_options
-        layout_options.lefdef_config.map_file = lym
-        layout_options.lefdef_config.macro_resolution_mode = 1
-        layout_options.lefdef_config.read_lef_with_def = False
-        layout_options.lefdef_config.lef_files = input_lefs.split(";")
-
-    view = pya.LayoutView()
-    view.load_layer_props(lyp)
-
-    if gds:
-        view.load_layout(input)
-    else:
-        view.load_layout(input, layout_options, lyt)
-    view.max_hier()
-    pixels = view.get_pixels_with_options(1000, 1000)
-
-    with open(output, "wb") as f:
-        f.write(pixels.to_png_data())
-
-    pya.Application.instance().exit(0)
-except Exception as e:
-    print(e)
-    pya.Application.instance().exit(1)
+from typing import Tuple
+
+import pya
+import click
+
+
+@click.command()
+@click.option("-o", "--output", required=True)
+@click.option(
+    "-l",
+    "--input-lef",
+    "input_lefs",
+    multiple=True,
+)
+@click.option(
+    "-T",
+    "--lyt",
+    required=True,
+    help="KLayout .lyt file",
+)
+@click.option(
+    "-P",
+    "--lyp",
+    required=True,
+    help="KLayout .lyp file",
+)
+@click.option(
+    "-M",
+    "--lym",
+    required=True,
+    help="KLayout .map (LEF/DEF layer map) file",
+)
+@click.argument("input")
+def render(
+    input_lefs: Tuple[str, ...],
+    output: str,
+    lyt: str,
+    lyp: str,
+    lym: str,
+    input: str,
+):
+    try:
+        gds = input.endswith(".gds")
+
+        # Load technology file
+        tech = pya.Technology()
+        tech.load(lyt)
+
+        layout_options = None
+        if not gds:
+            layout_options = tech.load_layout_options
+            layout_options.lefdef_config.map_file = lym
+            layout_options.lefdef_config.macro_resolution_mode = 1
+            layout_options.lefdef_config.read_lef_with_def = False
+            layout_options.lefdef_config.lef_files = list(input_lefs)
+
+        view = pya.LayoutView()
+        view.load_layer_props(lyp)
+
+        if gds:
+            view.load_layout(input)
+        else:
+            view.load_layout(input, layout_options, lyt)
+        view.max_hier()
+        pixels = view.get_pixels_with_options(1000, 1000)
+
+        with open(output, "wb") as f:
+            f.write(pixels.to_png_data())
+    except Exception as e:
+        print(e)
+        exit(1)
+
+
+if __name__ == "__main__":
+    render()
```

Comment:
should also pass `shlex.split(os.getenv("KLAYOUT_ARGV))"`, shouldn't you?

Reply:
Nope, no need- these ones are actual shell scripts so they can just normal arguments